chia/util/keychain.py

```diff
@@ -336,19 +345,28 @@ def _get_key_data(self, index: int, include_secrets: bool = True) -> KeyData:
         str_bytes = bytes.fromhex(read_str)
 
         pk_bytes: bytes = str_bytes[: G1Element.SIZE]
-        observation_root: ObservationRoot = G1Element.from_bytes(pk_bytes)
-        fingerprint = observation_root.get_fingerprint()
-        if len(str_bytes) == G1Element.SIZE + 32:
-            entropy = str_bytes[G1Element.SIZE : G1Element.SIZE + 32]
-        else:
+        if len(pk_bytes) == 32:
```

Comment:
since pk_bytes is generated from `str_bytes[: G1Element.SIZE]` isn't this `if` impossible to hit?

Reply:
if str_bytes is the secp case then it's only 32 bytes and `str_bytes[: G1Element.size]` will just return those 32 bytes. So the `if` is reached as expected. Could also take len(str_bytes) if you prefer though?

Reply:
Ohhh I see, I didn't know it worked that way.  I think it would be more clear to just switch on the length of the str_bytes yeah.  If 32, it's a launcher ID, if 48, its a pubkey only, if 80, it's a pubkey+privatekey